src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInListener.cs

```diff
@@ -60,11 +63,21 @@ public override void OnStartActivity(Activity activity, object payload)
                 return;
             }
 
+            // Create a brand new activity using ActivitySource
+            // and populate its tags correctly.
+            // This is double allocation of Activity! and probably bad for performance.
+            var newActivity = this.activitySource.StartActivity("HttpRequestIn", ActivityKind.Server);
+
             var request = context.Request;
 
             // see the spec https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/data-semantic-conventions.md
             var path = (request.PathBase.HasValue || request.Path.HasValue) ? (request.PathBase + request.Path).ToString() : "/";
 
+            if (newActivity != null)
+            {
+                newActivity.DisplayName = path;
+            }
+
```

Comment:
you can write this block as 

newActivity?.DisplayName = path;

Reply:
![image](https://user-images.githubusercontent.com/5232798/82606492-0623bc80-9b6c-11ea-91a7-001fec36dd27.png)
Gets this error. Need a newer c# version for it to work?

Reply:
add 
```Xml
    <LangVersion>8.0</LangVersion>  
```
to the csproj